sunpy/timeseries/sources/eve.py

```diff
@@ -2,19 +2,121 @@
 from pandas.io.parsers import read_csv
 from os.path import basename
-
+import sunpy.io
 from sunpy.timeseries.timeseriesbase import GenericTimeSeries
 from sunpy.util.metadata import MetaDict
-
+from pandas import DataFrame
 from astropy import units as u
+import numpy as np
+from datetime import datetime
+from matplotlib import dates
+from sunpy.time import parse_time
+
+__all__ = ['EVESpWxTimeSeries', 'ESPTimeSeries']
+
+
+class ESPTimeSeries(GenericTimeSeries):
+    """
+    SDO EVE/ESP Level1 data
+
+    The Extreme ultraviolet Spectro-Photometer (ESP) is an irradiance instrument 
+    which is part of the Extreme ultraviolet Variability Experiment (EVE) onboard 
+    SDO. ESP provides high time cadence (0.25s) EUV irradiance measurments in five 
+    channels, one soft X-ray and 4 EUV. The first four orders of the diffraction grating 
+    gives measurments centered on 18nm, 26nm, 30nm and 36nm. The zeroth order (obtained 
+    by 4 photodiodes) provides the soft X-ray measurments from 0.1-7nm. 
+
+    The ESP level 1 fits files are fully calibrated. The TimeSeries object created from
+    an ESP fits file will conatain 4 columns namely:
+
+        * 'QD' - sum of 4 quad diodes, this is the soft X-ray measurments 0.1-7nm
+        * 'CH_18' - EUV irradiance 18nm
+        * 'CH_26' - EUV irradiance 26nm
+        * 'CH_30' - EUV irradiance 30nm
+        * 'CH_36' - EUV irradiance 36nm        
+
+    References
+    ----------
+    * `SDO Mission Homepage <https://sdo.gsfc.nasa.gov/>`_
+    * `EVE Homepage <http://lasp.colorado.edu/home/eve/>`_
+    * `README ESP data <http://lasp.colorado.edu/eve/data_access/evewebdata/products/level1/esp/EVE_ESP_L1_V6_README.pdf>`_
+    * `ESP lvl1 data <http://lasp.colorado.edu/eve/data_access/evewebdata/misc/eve_calendars/calendar_level1_2018.html>`_
+    * `ESP instrument paper <https://doi.org/10.1007/s11207-009-9485-8>`_
+    
+    Notes
+    -----
+    The 36nm channel demonstrates a significant noise and it is not recommended to be 
+    used for short-time observations of solar irradiance.
+
+    """
+
+    _source = 'esp'
+
+    def peek(self, title='EVE/ESP Level1', **kwargs):
+
+        self._validate_data_for_ploting()
+
+        names = ('Flux \n 0.1-7nm', 'Flux \n 18nm', 'Flux \n 26nm', 'Flux \n 30nm', 'Flux \n 36nm')
 
-__all__ = ['EVESpWxTimeSeries']
+        figure = plt.figure()
+        axes = plt.gca()
+        axes = self.data.plot(ax=axes, subplots=True, sharex=True, **kwargs)
+        plt.xlim(self.data.index[0], self.data.index[-1])
+
+        axes[0].set_title(title)
+        for i, ax in enumerate(axes):
+            ax.set_ylabel(names[i])
+            ax.legend(loc='upper right')
+        axes[-1].set_xlabel('Time (UT) ' + str(self.data.index[0])[0:11])
+        axes[-1].xaxis.set_major_formatter(dates.DateFormatter('%H:%M'))
+        plt.tight_layout()
+        plt.subplots_adjust(hspace=0.05)
+
+        figure.show()
+
+    @classmethod
+    def _parse_file(cls, filepath):
```

Comment:
I would need to double check this, but I don't think you need this method for fits files.

Reply:
thanks for edits - have updated now and added a truncated version of fits file :P